### DISCUSSION TITLE
Filtering by multiple facets

### DISCUSSION THREAD

**Discussion by jonathanhuang13** [algolia/instantsearch#5341] [Q&A] [answered]
How do you implement a refinement list where each option does a filter on multiple facets?

For instance, I might have a list of courses I want to filter and each course has a start date and a deadline to enroll. I want a refinement widget that has options to filter by start date (i.e. in <2 weeks, in <3 weeks, etc.) and these filters will filter courses if both the start date and deadline are within the filter.

How do I do this using React Hooks? Reading the docs, I don't see a way to do this easily. In https://www.algolia.com/doc/guides/managing-results/refine-results/filtering/how-to/filter-by-date/, it says I can do something like

```
index.search('query', {
  filters: 'NOT date_timestamp:1538352000 TO 1540944000 AND NOT date_timestamp_2:1538352000 TO 1540944000'
}).then(({ hits }) => {
  console.log(hits);
});
```

to filter by date but what is `index` referring to here? In React InstantSearch hooks, I only have access to `searchClient` and the `refine` function.

Do I need to create a custom widget?

**Comment by Haroenv** [comment]
You may be looking for either NumericMenu / useNumericMenu or useRange

**Reply by jonathanhuang13** [reply]
Hmm how exactly will `useNumericMenu` or `useRange` help if I'm trying to filter by multiple attributes? These hooks seem to operate only on one attribute but I want to filter on multiple attributes with one click

**Comment by jonathanhuang13** [comment, marked as answer]
Looks like if you can't filter by multiple attributes. You have to put it into one

https://discourse.algolia.com/t/filtering-by-multiple-facets/16611/2

**Reply by Haroenv** [reply]
You can also use `useInstantSearch`' setIndexUiState to modify multiple states at once if those widgets are also mounted
